File: app/src/main/java/com/viola2013/popularmoviesapp/DetailsActivity.java
package com.viola2013.popularmoviesapp;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.text.ParseException;



/**
 * Created by Mervi on 2.5.2018.
 * Based on code from http://stackoverflow.com/a/11093572
 */
public class DetailsActivity extends AppCompatActivity {
    /**
     * To log any issue that might be worth logging
     */
    private final String LOG_TAG = DetailsActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_details);

        TextView tvOriginalTitle = findViewById(R.id.textview_original_title);
        ImageView ivPoster = findViewById(R.id.imageview_poster);
        TextView tvOverView = findViewById(R.id.textview_overview);
        TextView tvVoteAverage = findViewById(R.id.textview_vote_average);
        TextView tvReleaseDate = findViewById(R.id.textview_release_date);

        Intent intent = getIntent();
        Movie movie = intent.getParcelableExtra(getString(R.string.parcel_movie));

        tvOriginalTitle.setText(movie.getOriginalTitle());

        Picasso.with(this)
                .load(movie.getPosterPath())
                .resize(getResources().getInteger(R.integer.tmdb_poster_w185_width),
                        getResources().getInteger(R.integer.tmdb_poster_w185_height))
                .error(R.drawable.not_found)
                .placeholder(R.drawable.searching)
                .into(ivPoster);

        String overView = movie.getOverview();
        if (overView == null) {
            tvOverView.setTypeface(null, Typeface.ITALIC);
            overView = getResources().getString(R.string.no_summary_found);
        }
        tvOverView.setText(overView);
        tvVoteAverage.setText(movie.getDetailedVoteAverage());

        // First get the release date from the object - to be used if something goes wrong with
        // getting localized release date (catch).
        String releaseDate = movie.getReleaseDate();
        if(releaseDate != null) {
            try {
                releaseDate = DateTimeHelper.getLocalizedDate(this,
                        releaseDate, movie.getDateFormat());
            } catch (ParseException e) {
                Log.e(LOG_TAG, "Error with parsing movie release date", e);
            }
        } else {
            tvReleaseDate.setTypeface(null, Typeface.ITALIC);
            releaseDate = getResources().getString(R.string.no_release_date_found);
        }
        tvReleaseDate.setText(releaseDate);
    }
}

File: app/src/main/java/com/viola2013/popularmoviesapp/DetailsActivity.kt
package com.viola2013.popularmoviesapp

import android.os.Build
import android.os.Bundle
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import com.viola2013.popularmoviesapp.ui.screens.DetailsScreen
import com.viola2013.popularmoviesapp.ui.theme.PopularMoviesAppTheme

class DetailsActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val movie = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            intent.getParcelableExtra(getString(R.string.parcel_movie), Movie::class.java)
        } else {
            @Suppress("DEPRECATION")
            intent.getParcelableExtra(getString(R.string.parcel_movie))
        }

        if (movie == null) {
            finish()
            return
        }

        setContent {
            PopularMoviesAppTheme {
                DetailsScreen(
                    movie = movie,
                    onBackClick = { finish() },
                    onFavoriteClick = {
                        // Simulate favorite toggle
                        Toast.makeText(this, getString(R.string.favorite_added), Toast.LENGTH_SHORT).show()
                    }
                )
            }
        }
    }
}
